doc/spec_TransferElement.dox

```diff
@@ -263,9 +263,8 @@ This documnent is currently still **INCOMPLETE**!
 - \anchor transferElement_C_2 2. In doPostXxx no new ChimeraTK::runtime_error or ChimeraTK::logic_error are thrown. Exceptions that were risen in doPreXxx or doXxxTransferYyy, or that were received from the queue (see B.8.3) are rethrown. \ref transferElement_comment_C_2 "(*)"
   - 2.1 The TransferElement base class stores the caught exceptions in TransferElement::_activeException.
   - 2.2 postXxx() is re-throwing the exception after delegating to doPostXxx().
-  - \anchor transferElement_C_2_3 2.3 Decorators must delegate the TransferElement::_activeException to their target using TransferElement::setActiveException(). If the delegated exception is not nullptr,
-  the targets _active exception will be replaced by it, and the delegated exeption, which is handed by reference, is set to nullptr \ref transferElement_comment_C_2_3a "(*)". The target is now responsible for handling the exception. If the delegated exception already is nullptr, setActiveException() has no effect. A nullptr does not overwrite an active exception in the target \ref transferElement_comment_C_2_3b "(*)". \ref transferElement_comment_C_2_3c "(*)"
-    - 2.3.1 Decorators which are throwing themselves in doPreXxx(), before delegating to the target preXxx(), must remember this. If they did throw in doPreXxx(), then in doPostXxx() they do not swap the exception into the target and do not delegate to postXxx(). The exception is then re-thrown by the decorator's postXxx().
+  - \anchor transferElement_C_2_3 2.3 Decorators must delegate the TransferElement::_activeException to their target using TransferElement::setActiveException(). If the delegated exception is not nullptr, the targets _active exception will be replaced by it, and the delegated exeption, which is handed by reference, is set to nullptr \ref transferElement_comment_C_2_3a "(*)". The target is now responsible for handling the exception. If the delegated exception already is nullptr, setActiveException() has no effect. A nullptr does not overwrite an active exception in the target \ref transferElement_comment_C_2_3b "(*)". \ref transferElement_comment_C_2_3c "(*)"
```

Comment:
Typo, please correct:  `_active exception` -> ` _activeException`

Reply:
> and the delegated exeption, which is handed by reference, is set to nullptr`

I would change the wording:

> and the _activeException of the decorator is set to nullptr.

The "delegated exception" is in my opinion the exception itself, not the _activeException pointer to it. The exception is delegated to the target and hence not set to nullptr...

Reply:
> If the delegated exception already is nullptr, setActiveException() has no effect. A nullptr does not overwrite an active exception in the target.

This is duplicate, which IMO makes it harder to understand (because one assumes the second sentence has a different meaning). Also, I would also here replace "delegated exception" with "_activeException".

Reply:
And a general wording comment: I would prefer to avoid using "must" and "will" etc., because these words always imply a certain point of view. So I prefer:
> Decorators delegate the TransferElement::_activeException

and

> If the delegated exception is not nullptr, the targets _active exception is replaced by it

This makes the statements neutral and does not imply whether the reader is implementing a decorator, using one, or trying to understand what a decorator does to his backend implementation.

Reply:
I implemented all changes.
It would be nice to understand what is implemented in DeviceAccess already, and which part of the spec has to be followed when writing your own decorators, backends and transfer elements. But maybe the "must, will" etc. is not the right tool. I once worked on a document which started with an explanation "In this document should means this.., must means this, shall means something different". It was horrible to read and understand.

Reply:
"A does something" implies in the same way that A is responsible for doing so as "A must do something". IMO the second wording is just very hard to read if one is currently not interested in the requirements to A. Usually there are at least two sides for each statement. E.g. "A gives something to B if condition" implies many things: "A must give something to B under the condition", "B must receive something from A under the condition", "B can expect A to give something under the condition" (which is kind of equal to "A will give something to B under the condition"), and usually also that all this does not happen if the condition is not met. This kind of gets lost, without really adding something valuable.

Reply:
Right. We should use the simplest unambiguous wording. Adding extra words or fancy wording does not help.
In contrast to a letter or publication each sentence can, even should, always use the same structure. It does not have to be 'nice' to read, but as easy to understand as possible.